doc/iterations/2024.11/drag_and_drop_in_explorer.adoc

```diff
@@ -0,0 +1,35 @@
+= (M) Add support for drag and drop in the explorer
+
+== Problem
+
+The current approach allows drag-and-drop (DnD) operations from the explorer to a diagram representation.
+The goal is to enable DnD functionality within the explorer, with the resulting behavior being customizable through a specifier.
+
+== Key Result
+
+Enable users to drag and drop items inside the explorer, with the resulting behavior being customizable through a specifier.
```

Comment:
I'll just add a sentence to note that we want to drop tree items on another tree item and between other tree items